hibernate-hql-parser-antlr4poc/src/main/antlr/org/hibernate/query/parser/internal/hql/antlr/HqlParser.g4

```diff
@@ -285,18 +286,19 @@ likeEscape
 	;
 
 expression
-	: expression DOUBLE_PIPE expression		# ConcatenationExpression
-	| expression PLUS expression			# AdditionExpression
-	| expression MINUS expression			# SubtractionExpression
-	| expression ASTERISK expression		# MultiplicationExpression
-	| expression SLASH expression			# DivisionExpression
-	| expression PERCENT expression			# ModuloExpression
-	| MINUS expression						# UnaryMinusExpression
-	| PLUS expression						# UnaryPlusExpression
-	| literal								# LiteralExpression
-	| parameter								# ParameterExpression
-	| function								# FunctionExpression
-	| path									# PathExpression
+	: expression DOUBLE_PIPE expression			# ConcatenationExpression
+	| expression PLUS expression				# AdditionExpression
+	| expression MINUS expression				# SubtractionExpression
+	| expression ASTERISK expression			# MultiplicationExpression
+	| expression SLASH expression				# DivisionExpression
+	| expression PERCENT expression				# ModuloExpression
+	| MINUS expression							# UnaryMinusExpression
+	| PLUS expression							# UnaryPlusExpression
+	| literal									# LiteralExpression
+	| parameter									# ParameterExpression
+	| function									# FunctionExpression
+	| path										# PathExpression
+	| LEFT_PAREN selectStatement RIGHT_PAREN	# Subquery
```

Comment:
selectStatement is the wrong sub-rule here.  It should be querySpec.  A sub-query cannot have an order-by clause